doc/http-grpc-status-mapping.md

```diff
@@ -0,0 +1,30 @@
+# HTTP to gRPC Status Code Mapping
+
+Since intermediaries are a common part of HTTP infrastructure some responses to
+gRPC requests may be received that do not include the grpc-status header. In
+some cases mapping error codes from an intermediary allows the gRPC client to
+behave more appropriately to the error situation without overloading the
+semantics of either error code.
+
+This table is to be used _only_ for clients that received a response that did
+not include grpc-status. If grpc-status was provided, it _must_ be used. Servers
+_must not_ use this table to determine an HTTP status code to use; the mappings
+are neither symmetric nor 1-to-1.
+
+| HTTP Status Code           | gRPC Status Code   |
+|----------------------------|--------------------|
+| 400 Bad Request            | INTERNAL           |
```

Comment:
407 seems a bit strange, given that we don't currently support client-side proxies. It also might be confusing as UNAUTHENTICATED because clients would generally assume that the end-server failed the request and probably wouldn't notice the Proxy-Authenticate. There are subtleties, so I think I'd favor leaving it off for the moment.

500 seems more appropriate as UNKNOWN. Most commonly that is due to an exception on the server, which we would generally convert to UNKNOWN since it is in a different "error space."